packages/app/src/components/nominationInfo/EditCard.js

```diff
@@ -1,59 +1,117 @@
+      ...base,
+      border: '0.1px solid #d1d1d1',
+      boxShadow: 'none',
+      '&:hover': {
+        border: '1px solid var(--brand)',
+      },
+      width: '50%',
+      minHeight: '40px',
+      height: '40px',
+      cursor: 'pointer',
+    }),
+    valueContainer: (provided) => ({
+      ...provided,
+      height: '40px',
+      padding: '0 6px',
+    }),
+    input: (provided) => ({
+      ...provided,
+      margin: '1px',
+    }),
+    indicatorsContainer: (provided) => ({
+      ...provided,
+      height: '40px',
+    }),
+  };
 
   return (
     <div className={styles.card}>
       <div className={[styles.gridContainer, styles.content].join(' ')}>
-        {props.keys.map(label => {
+        {props.keys.map((label) => {
           switch (true) {
             // Render our titles in form
             case props.titleLabels.includes(label):
-              return <Title label={label} />
+              return <Title label={label} key={label} />;
             // Render dates
             case props.editableDates.includes(label):
               return (
-                <div>
+                <div key={label}>
                   <label className={styles.label}>{label}</label>
                   {/* Controller Component needed for React Hook Form and React Datepicker Integration*/}
                   <Controller
                     name={label}
                     control={props.control}
                     defaultValue={
-                      // from useState hooks above
-                      String(admissionDate) != 'Invalid Date' && String(dischargeDate) != 'Invalid Date'
+                      String(props.formData['Admission Date']) !==
+                        'Invalid Date' &&
+                      String(props.formData['Discharge Date']) !==
+                        'Invalid Date'
                         ? label === 'Admission Date'
-                          ? new Date(admissionDate)
-                          : new Date(dischargeDate)
+                          ? new Date(props.formData['Admission Date'])
+                          : new Date(props.formData['Discharge Date'])
                         : null
                     }
                     render={({ field: { onChange, value } }) => {
                       return (
                         <DatePicker
-                          selected={value}
-                          onChange={onChange}
-                          dateFormat='MM/dd/yyyy'
+                          selected={new Date(value)}
+                          onChange={(date) => {
+                            if (!date) {
+                              onChange(new Date());
+                            } else {
+                              const yearLength = date.getFullYear().toString()
+                                .length;
+                              if (yearLength === 4) {
+                                onChange(new Date(date));
+                              }
```

Comment:
Ditto all the comments above. Maybe the function could be written once, imported into all the components where we need it, and we could pass in the `date` as well as a callback?